File: current_session/python/88.py
'''
88
'''
from typing import List
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        c1, c2 = 0, 0
        if n == 0: return
        tmp = []
        while c1 < m and c2 < n:
            if nums1[c1] <= nums2[c2]:
                tmp.append(nums1[c1])
                c1 += 1
            else:
                tmp.append(nums2[c2])
                c2 += 1
        
        while c2 == n and c1 < m:
            tmp.append(nums1[c1])
            c1 += 1
        while c1 == m and c2 < n:
            tmp.append(nums2[c2])
            c2 += 1
        # print(tmp, nums1)
        for i in range(m+n):
            nums1[i] = tmp[i]
        return

File: current_session/c++/88.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
  void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    if (n==0) return;
    if (m==0) {
      for (int i=0; i<nums1.size(); ++i) nums1[i] = nums2[i];
      return;
    }
    int p1 = m-1;
    int p2 = n-1;
    int x = m+n-1;
    while (p2>=0 || p1>=0) {
      if (p2<0) {
        break;
      } else if (p1>=0 && p2>=0 && nums1[p1] >= nums2[p2]) {
        nums1[x] = nums1[p1];
        --x;
        --p1;
      } else {
        nums1[x] = nums2[p2];
        --x;
        --p2;
      } 
    }
    return;
  }
};